### DISCUSSION TITLE
How to build ROCm for target gfx803 (rx580)

### DISCUSSION THREAD

**Discussion by fzadikian** [rocm-arch/rocm-arch#950] [Support] [answered]
Hello, I've tried running `env ROC_USE_PRE_VEGA=1 ROCM_HOME=/opt/rocm/ AMDGPU_TARGET=gfx803 pacaur -S rocm-hip-sdk rocm-opencl-sdk` but if the packages get downloaded from arch4edu/community-testing, they aren't built for my target (since it's just binaries). Then, both packages are now missing from the AUR (same with rocm-hip-runtime and rocm-opencl-runtime). How could I use this repository to accomplish my goal? Thank you.

Related: https://github.com/rocm-arch/rocm-arch/issues/866

**Comment by tpkessler** [comment]
Packages are built with the default flags from AMD that don't include gfx803. Which parts of the ROCm stack would you like to use? OpenCL should be possible, pytorch would not work.

**Reply by fzadikian** [reply]
First off, sorry that I started this as an issue, I had forgotten discussions existed. Secondly, I would like to use pytorch. Actually, I already got it working on a docker container, using [this](https://github.com/xuhuisheng/rocm-gfx803), so I guess it shouldn't really be impossible. I would like to run it without using docker on Artix Linux. I already made good progress, apparently the ROCm stack is working, but the pytorch binaries from xuhuisheng are not. I even tried building pytorch myself and still nothing. This is the error I got:
```
/usr/include/c++/12.2.0/bits/stl_vector.h:1123: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = const char*; _Alloc = std::allocator<const char*>; reference = const char*&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Abortado
```
Related issue with more information: https://github.com/xuhuisheng/rocm-gfx803/issues/26
Thank you for your help!

**Reply by jellersby** [reply]
This might help: https://github.com/RadeonOpenCompute/ROCR-Runtime/issues/153

**Comment by fzadikian** [comment, marked as answer]
Just in case anyone comes across this, the ROCm stack from [community] is working for gfx803. The new pytorch package from [community-testing] is working as well. The python-torchvision-rocm and python-torchaudio-rocm PKGBUILDs from the AUR are working as well.

**Reply by HeartFire01** [reply]
> Just in case anyone comes across this, the ROCm stack from [community] is working for gfx803. The new pytorch package from [community-testing] is working as well. The python-torchvision-rocm and python-torchaudio-rocm PKGBUILDs from the AUR are working as well.

Thanks for the info, but i have a problem my `rocminfo `only shows the CPU...Anyone how to make GPU is detected?...

**Reply by xuhuisheng** [reply]
@HeartFire01

It seems you met the PCIe Atomic issues, It said that you must make sure both CPU and motherboard support PCIe atomic feature, otherwise the gfx803 cannot be added to the ROCm device list.

To confirm that, execute `dmesg |grep kfd` could see something like `kfd: skipped device [...], PCI rejects atomics`.

And yes, only gfx803 required PCIe atomic feature. vega, mi100 didn't need PCIe atomic feature, RDNA2 can upgrade firmware to skip PCIe atomic requirement. Only gfx803 hang for it.

If you have enough time to dig it, please read this:
<https://github.com/RadeonOpenCompute/ROCm/tree/290091946ff417b6d402ba5eb2be62bb27e556c1#hardware-and-software-support>
